diff -Naur '--exclude-from=diff-ignore' enigma2-nightly.org/lib/base/eptrlist.h enigma2-nightly/lib/base/eptrlist.h
--- enigma2-nightly.org/lib/base/eptrlist.h	2013-03-03 12:56:00.845947716 +0100
+++ enigma2-nightly/lib/base/eptrlist.h	2013-03-03 13:14:47.085900352 +0100
@@ -175,7 +175,7 @@
 	{
 		// added a new item to the list... in order
 		// returns a iterator to the new item
-		return insert( std::lower_bound( std::list<T*>::begin(), std::list<T*>::end(), e, less()), e );
+		return this->insert( std::lower_bound( std::list<T*>::begin(), std::list<T*>::end(), e, less()), e );
 	}
 
 };
