diff --git a/webinterface/src/LICENSE b/webinterface/src/LICENSE
index b81fd7d..3685d31 100644
--- a/webinterface/src/LICENSE
+++ b/webinterface/src/LICENSE
@@ -10,8 +10,3 @@ is licensed by Dream Multimedia GmbH.
 This plugin is NOT free software. It is open source, you are allowed to
 modify it (if you keep the license), but it may not be commercially 
 distributed other than under the conditions noted above.
-
-As an exception regarding Modifcations, you are NOT permitted to remove
-any geniune checks implemented in this Software or change them for means of disabling 
-or working around the genuine checks, unless the change has been explicitly permitted 
-by the original Author(s).
